### PR TITLE
feat: minikube k8s can pull from ECR

### PR DESCRIPTION
**Pull request by artsyjian**
Mainly to close out the open comments in https://github.com/artsy/hokusai/pull/400.

- Let `hokusai staging refresh` test see to `successfully rolled out` output. ([Comment addressed](https://github.com/artsy/hokusai/pull/400#discussion_r1463428499) from previous PR)
- Let CI `test_integration` step and `scripts/integration_test_local.sh` create minikube/k8s regcred for pulling image from ECR.
- Rename `make dependencies` to `make hokusai`, since Poetry installs Hokusai. ([Comment addressed](https://github.com/artsy/hokusai/pull/400#discussion_r1459270493) from previous PR)
- Rename `test_git_repo_name` fixture. ([Comment addressed](https://github.com/artsy/hokusai/pull/400#discussion_r1463383716) from previous PR)

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -106,22 +107,34 @@ jobs:
...
+              docker login --username AWS \
+                --password-stdin ${aws_account_id}.dkr.ecr.${AWS_DEFAULT_REGION}.amazonaws.com
       - run:
-          name: dump cluster-info
+          name: setup regcred for staging minikube/k8s
           command: |
-            kubectl cluster-info
-            kubectl get pods --all-namespaces
+            kubectl --context staging create secret generic regcred \
+              --from-file=.dockerconfigjson=${HOME}/.docker/config.json \
```
Comment (cavvia) [inline .circleci/config.yml:122]: I was a little confused as to where this file comes from, but it seems from the [doc](https://kubernetes.io/docs/tasks/configure-pod-container/pull-image-private-registry/#registry-secret-existing-credentials) that it's placed there by `docker login` in the previous step. 👍🏽

File: test/integration/test_staging.py [hunk 2]
```diff
@@ -62,22 +62,21 @@ def it_reports_no_changes():
...
+      text=True,
+      timeout=30
+    )
+    if resp.returncode != 0:
+      print(resp.stderr)
+    assert resp.returncode == 0
+    assert 'Refreshing hokusai-integration-test-web' in resp.stdout
+    assert 'Waiting for refresh to complete' in resp.stdout
+    assert 'Waiting for deployment "hokusai-integration-test-web" rollout to finish' in resp.stdout
+    assert 'deployment "hokusai-integration-test-web" successfully rolled out' in resp.stdout
```
Comment (cavvia) [inline test/integration/test_staging.py:79]: Nice!

File: .circleci/config.yml (hunk 1 above)
Comment (artsyjian) [reply .circleci/config.yml:122]: yeah i think so.

**Review by cavvia — APPROVED** [review]
Great that you managed to get this working end to end with a successful image pull inside minikube. Looks like quite a tricky process to get the credentials in place. Assuming there aren't any security concerns with the credentials in minikube it looks good to me!

File: scripts/integration_test_local.sh [hunk 3]
```diff
@@ -15,23 +34,82 @@ function start_minikube(){
...
+}
+
+function restore_kube_config() {
+  echo "Restoring $KUBE_CONFIG_PATH ..."
+  mv "$KUBE_CONFIG_BACKUP_PATH" "$KUBE_CONFIG_PATH"
+}
+
+function create_regcred() {
+  # create ecr image pull secret in minikube/k8s
+  aws ecr get-login-password --region "$AWS_REGION" | \
```
[the remark this answers is not in the document]
Comment (artsyjian) [reply scripts/integration_test_local.sh:82]: the steps are very env-specific, when i attempted to consolidate into one script, it was a lot of "if i am running in Circle, do this, if local, do that, ...", so i aborted.

[the remark this answers is not in the document]
Comment (cavvia) [reply scripts/integration_test_local.sh:82]: Ah I see this is for manual run on local, got it.

**Pull request merged**